### PR TITLE
Remove system properties from merged model config

### PR DESCRIPTION
**Pull request by piotrp**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: extensions-api/src/main/scala/pl/touk/nussknacker/engine/modelconfig/ModelConfigLoader.scala [hunk 1]
```diff
@@ -100,8 +101,15 @@ abstract class ModelConfigLoader extends Serializable with LazyLogging {
...
       and have baseUrl taken from application config
      */
-    // We want to respect overrides (like system properties) and standard fallbacks (like reference.conf)
-    ConfigFactory.load(classLoader, withFallbackToDefaults(inputConfig, classLoader))
+    // We want to respect overrides (via environment variables) and standard fallbacks (like reference.conf).
+    // Code below works like 'ConfigFactory.load(ClassLoader,Config)' but it:
+    // - doesn't include system properties
+    // - always resolves overrides from environment variables (works as if 'config.override_with_env_vars' was always enabled')
+    ConfigFactory
+      .systemEnvironmentOverrides()
```
Comment (arkadius) [inline extensions-api/src/main/scala/pl/touk/nussknacker/engine/modelconfig/ModelConfigLoader.scala:109]: IMO we should remove these overrides for the sake of clarity that it is done only on the designer stage but is not mandatory for now, feel free to omit this.

**Review by arkadius — APPROVED** [review]

File: extensions-api/src/main/scala/pl/touk/nussknacker/engine/modelconfig/ModelConfigLoader.scala (hunk 1 above)
Comment (piotrp) [reply extensions-api/src/main/scala/pl/touk/nussknacker/engine/modelconfig/ModelConfigLoader.scala:109]: Ok, I will remove them

File: extensions-api/src/main/scala/pl/touk/nussknacker/engine/modelconfig/ModelConfigLoader.scala [hunk 2]
```diff
@@ -100,8 +101,11 @@ abstract class ModelConfigLoader extends Serializable with LazyLogging {
       service1Url: ${baseUrl}/service1
       and have baseUrl taken from application config
      */
-    // We want to respect overrides (like system properties) and standard fallbacks (like reference.conf)
-    ConfigFactory.load(classLoader, withFallbackToDefaults(inputConfig, classLoader))
+    // We want to respect standard fallbacks (like reference.conf), but leave resolution of overrides and environment
```
Comment (arkadius) [inline extensions-api/src/main/scala/pl/touk/nussknacker/engine/modelconfig/ModelConfigLoader.scala:104]: Actually, resolution of environment variables is sometime that here as well (behind `resolve(ConfigResolveOptions.defaults)`. We only skip system properties resolution and environment variables override mechanism

Comment (piotrp) [reply extensions-api/src/main/scala/pl/touk/nussknacker/engine/modelconfig/ModelConfigLoader.scala:104]: Can you give an example? I tried to write a test for that (in `DefaultModelConfigLoaderTest`) and it fails in the call you mentioned with `Could not resolve substitution to a value: ${PATH}`.

Modified defaultModelConfig.conf:
```conf
{
  property1: shouldBeOverridenToo
  property2: ${property1}Suffix
  otherProperty: true
  envPathProperty: ${PATH}
}
```

Test:
```scala
  test("should resolve environment variables") {
    val config = LocalModelData(inputConfig, List.empty).modelConfig
    val envPath = System.getenv("PATH")

    envPath shouldNot be(null)
    config.getString("envPathProperty") shouldBe envPath
  }
```

Comment (arkadius) [reply extensions-api/src/main/scala/pl/touk/nussknacker/engine/modelconfig/ModelConfigLoader.scala:104]: strange, I do the same on your branch and it works for me :) https://github.com/TouK/nussknacker/pull/5421/files

**Pull request merged**